src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/SwiftPhysicalLowering.cs

```diff
@@ -0,0 +1,196 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Internal.TypeSystem;
+
+namespace Internal.JitInterface
+{
+    public static class SwiftPhysicalLowering
+    {
+        public abstract record Lowering
+        {
+            private Lowering() { }
+
+            public sealed record InvalidType : Lowering;
+
+            public sealed record PassByRef : Lowering;
+
+            public sealed record PrimitiveSequence(ImmutableArray<CorInfoType> Sequence) : Lowering
```

Comment:
What are we getting out of these being `record`? We don't count every byte of the compiler size, but `record` is a death by a thousand papercuts and I don't think we use it anywhere in the compiler.

Reply:
I personally like using records to model a discriminated-union-style structure. I'll change this to directly use the recommended JITInterface type.